lib/chef/provider/remote_file/http.rb

```diff
@@ -122,6 +122,10 @@ def etag_from(response)
 
         def http_client_opts
           opts = {}
+          # For http request retry and retry_delay was coming from config (default values i.e. retry: 5, retry_delay: 2)
+          # Added retries, retry_delay options to set "retries" and "retry_delay" properly
+          opts[:retries] = new_resource.retries
+          opts[:retry_delay] = new_resource.retry_delay
```

Comment:
instead of this there should be a `property :http_options` hash added to the remote_file resource and that should be merged with the defaults here in http_client_opts

Reply:
Added `property :http_options` , please have look.